mirrord/agent/src/file.rs

```diff
@@ -569,6 +573,31 @@ impl FileManager {
         .map_err(ResponseError::from)
     }
 
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub(crate) fn xstatfs(&mut self, fd: u64) -> RemoteResult<XstatFsResponse> {
+        let target = self
+            .open_files
+            .get(&fd)
+            .ok_or(ResponseError::NotFound(fd))?;
+
+        let statfs = match target {
+            RemoteFile::File(file) => nix::sys::statfs::fstatfs(file).unwrap(),
+            RemoteFile::Directory(path) => nix::sys::statfs::statfs(path).unwrap(),
+        };
+
+        let metadata = FsMetadataInternal {
```

Comment:
If this conversion can't fail, what do you think of moving it to a `From` implementation?